src/charm_state.py

```diff
@@ -24,6 +73,8 @@ class CharmState:
 
     def __init__(
         self,
+        *,
+        flask_config: dict[str, int | str] | None = None,
```

Comment:
What about using a `TypedDict` so that it is clear which keys are available?

Reply:
There's no need to access each individual fields inside the `flask_config`, so a simple dict is sufficient.